src/equations/numerical_fluxes.jl

```diff
@@ -53,19 +53,44 @@ struct FluxRotated{NumericalFlux}
   numerical_flux::NumericalFlux
 end
 
-@inline function (flux_rotated::FluxRotated)(u_ll, u_rr, normal_vector::AbstractVector, equations)
+@inline function (flux_rotated::FluxRotated)(u_ll, u_rr, normal, equations::AbstractEquations{2})
   @unpack numerical_flux = flux_rotated
 
-  norm_ = norm(normal_vector)
-  # normalize the vector without using `normalize` since we need to multiply by the `norm_` later
-  normal = normal_vector / norm_
+  norm_ = norm(normal)
+  # Normalize the vector without using `normalize` since we need to multiply by the `norm_` later
+  normal_normalized = normal / norm_
 
-  u_ll_rotated = rotate_to_x(u_ll, normal, equations)
-  u_rr_rotated = rotate_to_x(u_rr, normal, equations)
+  u_ll_rotated = rotate_to_x(u_ll, normal_normalized, equations)
+  u_rr_rotated = rotate_to_x(u_rr, normal_normalized, equations)
 
   f = numerical_flux(u_ll_rotated, u_rr_rotated, 1, equations)
 
-  return rotate_from_x(f, normal, equations) * norm_
+  return rotate_from_x(f, normal_normalized, equations) * norm_
+end
+
+
+@inline function (flux_rotated::FluxRotated)(u_ll, u_rr, normal, equations::AbstractEquations{3})
+  @unpack numerical_flux = flux_rotated
+
+  norm_ = norm(normal)
+  # Normalize the vector without using `normalize` since we need to multiply by the `norm_` later
+  normal_normalized = normal / norm_
+
+  # Some vector that can't be identical to normal (unless normal == 0)
+  tangent1 = SVector(normal[2], normal[3], -normal[1])
+  # Orthogonal projection
+  tangent1 -= dot(normal_normalized, tangent1) * normal_normalized
+  tangent1 = normalize(tangent1)
+
+  # Third orthogonal vector
+  tangent2 = normalize(cross(normal, tangent1))
```

Comment:
@andrewwinters5000 Can there be any problems with this approach? Instead of projecting `Ja^2`, we chose some (arbitrary) vector that is different from `normal` and projected this one.
This allows for cleaner code, as we don't have to pass around the tangent vectors, and don't have to dispatch to distinguish between `FluxRotated` and any other flux (that doesn't take tangent vectors as argument).

We want to make sure that this doesn't cause any problems that we don't see right now.

Reply:
Off the top of my head this should be kosher for most situations and "nice" meshes. I looked in "I Do Like CFD"  documents and he recommends the following strategy (taken from fortran code and in his notation the vector `l`is your `tangent1`) 

! l = (lx.ly,lz)
!
! We first carefully choose one tangent vector: l = (lx,ly,lz).
!  E.g., if n = (1,0,0), then, (0,-nz,ny) = (0,0,0), which is a problem.
!        To avoid such a situation, we choose the 2D tangential vector
!        having the maximum length.

     tempx = ny*ny + nz*nz
     tempy = nz*nz + nx*nx
     tempz = nx*nx + ny*ny

     if     ( tempx >= tempy .and. tempx >= tempz ) then
       lx =  zero
       ly = -nz
       lz =  ny
     elseif ( tempy >= tempx .and. tempy >= tempz ) then
       lx = -nz
       ly =  zero
       lz =  nx
     elseif ( tempz >= tempx .and. tempz >= tempy ) then
       lx = -ny
       ly =  nx
       lz =  zero
     else
      ! Impossible to happen
      write(*,*) "subroutine inviscid_roe: Impossible to happen. Please report the problem."
      stop
     endif

!     Make it the unit vector.
      temp = sqrt( lx*lx + ly*ly + lz*lz )
       lx = lx/temp
       ly = ly/temp
       lz = lz/temp

Reply:
Thanks for looking into it.
Why do we want to choose a 2D tangential vector where one coordinate equals zero?
In our case, we also can't get a zero vector, since we don't choose any coordinate to be zero.